draco/asp/soft.lp

```diff
@@ -626,145 +641,214 @@ preference(summary_rect,M) :-
+    not helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_continuous_color) Continuous color for summary tasks.
 preference(summary_continuous_color,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,color),
-    not helper(mark_encoding_discrete_or_binned,_,E).
+    not helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_continuous_size) Continuous size for summary tasks.
 preference(summary_continuous_size,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,size),
-    not helper(mark_encoding_discrete_or_binned,_,E).
+    not helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_continuous_text) Continuous text for summary tasks.
 preference(summary_continuous_text,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,text),
-    not helper(mark_encoding_discrete_or_binned,_,E).
+    not helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_discrete_x) Discrete x for summary tasks.
 preference(summary_discrete_x,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,x),
-    helper(mark_encoding_discrete_or_binned,_,E).
+    helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_discrete_y) Discrete y for summary tasks.
 preference(summary_discrete_y,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,y),
-    helper(mark_encoding_discrete_or_binned,_,E).
+    helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_discrete_color) Discrete color for summary tasks.
 preference(summary_discrete_color,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,color),
-    helper(mark_encoding_discrete_or_binned,_,E).
+    helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_discrete_size) Discrete size for summary tasks.
 preference(summary_discrete_size,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,size),
-    helper(mark_encoding_discrete_or_binned,_,E).
+    helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_discrete_shape) Discrete shape for summary tasks.
 preference(summary_discrete_shape,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,shape),
-    helper(mark_encoding_discrete_or_binned,_,E).
+    helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
 
 % @soft(summary_discrete_text) Discrete text for summary tasks.
 preference(summary_discrete_text,E) :-
     attribute(task,root,summary),
     attribute((encoding,channel),E,text),
-    helper(mark_encoding_discrete_or_binned,_,E).
+    helper(mark_encoding_discrete_or_binned,_,E),
+    helper(enc_interesting,E).
```

Comment:
It feels strange to me to add interestingness to all these preferences. Won't that break if we don't have interestingness?

Reply:
with the changes, these presences surely have different meaning now. However I also checked that the current version is what meant by draco1 ("for summary task, the interested column are preferred to be discrete text. I also changed the test cases for them to not be broken.

Reply:
I think we should have simpler constraints when possible. Would it hurt to remove the restriction to interesting fields from the constraints?

Reply:
I think we should have simpler constraints when possible. Do we use interestingness only with task constraints? Would it hurt to remove the restriction to interesting fields from constraints that don't involve tasks?